### ISSUE TITLE
Receiving "Error: Index passed to line/column function does not keep with order" on invalid html

### ISSUE THREAD

**Issue by forrestab** [htmllint/htmllint#225]
I have narrowed this error down to the following code inside of an angular template file:

```html
<p>
    <h4>Something</h4>
</p>
```

This is for 0.7.0 and thanks for this tool.

**Comment by AlexLerman** [comment]
I've seen the same thing when there's no ending tag (just passed as a string):

`<div><p>Anything</p><div>`

Should not throw an error.

**Comment by mlochbaum** [comment]
This looks related to a lot of the things in #176.
